Sanity check for merge into v3.x.x

mihails-strasuns-sociomantic opened the pull request.

mathias-lang-sociomantic committed:
Provide the ability to specify custom layouts from the config

This new feature allow to cover the use cases that where overseen during the transition to newLogger.
It will also allow to remove the ConsoleLayout and FileLayout template parameters in the future.

mathias-lang-sociomantic committed:
Add support for makeLayout in LogExt/DaemonApp

mihails-strasuns-sociomantic committed:
Clean release notes

david-eckardt-sociomantic committed:
Use inotify1_init rather than inotify_init + fcntl

The point of inotify1_init1 is to avoid fcntl calls like these, in order to
make the code simpler and avoid a race condition if close-on-exec should be
set for the file descriptor.

mihails-strasuns-sociomantic committed:
Adjust few wrong copyright statements

david-eckardt-sociomantic committed:
Add global open_with_close_on_exec variable

This variable is read by all ocean functions that obtain file descriptors
from the system. If it is true then these functions enable the
close-on-exec option.

david-eckardt-sociomantic committed:
Use core.sys.linux.sys.signalfd

mihails-strasuns-sociomantic committed:
Add ocean.task.util.Event

mathias-lang-sociomantic committed:
log.Config: Add FILE and LINE parameter to throw new Exception

In D1, it's not here by default.

mihails-strasuns-sociomantic committed:
Specify exception type thrown by `awaitRunningTasks`

mihails-strasuns-sociomantic committed:
Explain why awaitRunningTasks leaves one active

Commit:
Task: Add missing protection attributes

Commit:
Task: Fix documentation in assignTo()

Commit:
Task: Add protection attribute to static ctor

Commit:
Task: Add documentation to static constructor

Commit:
Task: Fix documentation headers style

mihails-strasuns-sociomantic committed:
Handle nested Contiguous fields in deserializer

Fixed #87

mihails-strasuns-sociomantic committed:
Merge tag 'v2.7.1' into v2.x.x

mihails-strasuns-sociomantic committed:
Merge tag 'v2.7.2' into v2.x.x

mihails-strasuns-sociomantic committed:
Start new ocean.meta package

Started by splitting qualifier related transition templates from
`ocean.transition`.

These utilities are heavily used even among most primitive of type traits -
separating them into own module with no imports help to simplify import chain
for all other modules as `ocean.transition` also defined more heavy-weight
stuff.

mihails-strasuns-sociomantic committed:
Define most basic set of type traits

The `Basic` module contains basic non-recursive type traits (mostly adapted from
old `ocean.core.Traits_tango` module) that are needed to distinguish between
different kinds of types.

llucax committed:
travis: Push converted tags to ocean-d2 repo

Fixes #84.

mihails-strasuns-sociomantic committed:
Implement `ReduceType` type utility

Allows to reduce arbitrary type to a single value in a generic form by
recursively stripping its signature into smaller bits and processing with
user-supplied "reducer".

mihails-strasuns-sociomantic committed:
Implement some of indirection traits with reducer

Re-implements few of existing traits from `ocean.core.Traits` using new
`ReduceType` facility while keeping original unit tests to ensure new
implementation is sufficietly feature complete.

llucax committed:
travis: Add dub.sdl to ocean-d2

llucax committed:
travis: Use "generic" language

This is an undocumented Travis feature, using "generic" as language
doesn't do any previous setup.

See https://github.com/travis-ci/travis-ci/issues/4895#issuecomment-150703192

llucax committed:
travis: Make config file a bit more compact

Geod24 committed:
Formatter: Deprecate the size_t returning delegate

The return value is not used and the compatibility with Layout_tango is not needed.
Using void as a return type simplifies user code.

mihails-strasuns-sociomantic committed:
Allow default seed/accumulate for ReduceType

ReduceType is primarily intended as a way to implement complex type traits
which answer simple true/false question. All such traits will use same
accumulator implementation that applies logical OR to all individual results,
thus it is practical to make it default.

mihails-strasuns-sociomantic committed:
Release notes for ocean.meta

mathias-lang-sociomantic committed:
Move error-suppressing static asserts to unittest

The is no advantage in having them within the body, are unittests for ocean (and other applications) are always run.
However it does suppress error during development and lead to extra steps while debugging some core code (e.g. formatter).

mihails-strasuns-sociomantic committed:
Deprecate contiguous serializer traits

Those originated in over-generalized assumption that new framework will
be use to adjust all over serializers but it proved to be impactical for
the time.

At the same time extra compile-time reflection complicated code
structure consierably and resulted in suppression of relevant compiler
messages. It is better to deprecate/remove it for now and reconsider if
standardization of serializer is to happen again.

mathias-lang-sociomantic committed:
Formatter: Deprecate accidentally-public sink type

mathias-lang-sociomantic committed:
Deprecate ocean.io.device.{SerialPort,ProgressFile}

Because of EUSERNOTFOUND

Commit:
Allow core.Traits.hasMethod to handle interfaces

Commit:
Merge tag 'v2.6.7' into v2.x.x

v2.6.7

Commit:
Merge tag 'v2.7.3' into v2.x.x

v2.7.3

mihails-strasuns-sociomantic committed:
Clean release notes

mihails-strasuns-sociomantic commented:
@mathias-lang-sociomantic I have got plenty of merge conflicts including ones for logger and seems like resolved some of them wrong as formatting test is failing now. Could you have a look please?

mathias-lang-sociomantic commented:
Looks like I broke something :(

mathias-lang-sociomantic commented:
Fix is in https://github.com/sociomantic-tsunami/ocean/pull/120
I tried cherry-picking that commit on top of your merge, test passed.

Geod24 committed:
Formatter: toString overloads are not recognized anymore

This regression was introduced in #112. While changing the type internally used by the formatter,
it affected which toString overload can be used.
We now correctly recognize both toString overloads and issue a 'deprecation' message when size_t is encountered.

mihails-strasuns-sociomantic committed:
Merge tag 'v2.8.1' into v2.x.x

mihails-strasuns-sociomantic committed:
Merge branch 'v2.x.x' into v3.x.x
